Character finder sprint5

imadrig opened the pull request:
This pull request includes updates to the UI to better match the final design -- it does not match fully yet. Pagination has been implemented using react-paginate (npm was installed for this) and the search feature has been updated to correct the bugs with the arrows and backspace. Still working on the search feature to ensure it can search for entities/hex/css codes and so on.

Commit:
Updates UI, adds pagination using react-paginate, updates search feature

Commit:
updates package-lock.json to match whats in development branch

Commit:
updates package-lock json again to match what's in dev branch

Commit:
Updates package-lock-json again to match dev branch

joekotvas commented on fetools-app/src/components/CharacterFinderComponents/CharacterCard.jsx:
```diff
@@ -9,21 +9,44 @@ const CharacterCard = ({
+        <div className="flex flex-wrap">
+          <div className="w-full lg:w-1/2">
+            <p className="pt-2 text-xs text-left uppercase">
+              unicode: <p><span className="font-bold">{unicode || " -- "}</span></p>
```
```suggestion
              unicode: <p><span className="font-bold normal-case">{unicode || " -- "}</span></p>
```

joekotvas commented on fetools-app/src/components/CharacterFinderComponents/CharacterCard.jsx:
```diff
@@ -9,21 +9,44 @@ const CharacterCard = ({
+          
+          <div className="w-full lg:w-1/2">
+            <p className="pt-2 text-xs text-left uppercase">
+              html code: <p><span className="font-bold flex flex-wrap">{htmlcode || " -- "}</span></p>
```
```suggestion
              html code: <p><span className="font-bold normal-case break-all">{htmlcode || " -- "}</span></p>
```

joekotvas commented on fetools-app/src/components/CharacterFinderComponents/CharacterCard.jsx:
```diff
@@ -9,21 +9,44 @@ const CharacterCard = ({
+          <div className="w-full lg:w-1/2">
+            <p className="pt-2 text-xs text-left uppercase">
+              html entity:
+              <p><span className="font-bold">
```
```suggestion
              <p><span className="font-bold normal-case">
```

joekotvas commented on fetools-app/src/components/CharacterFinderComponents/CharacterCard.jsx:
```diff
@@ -9,21 +9,44 @@ const CharacterCard = ({
+          </div>
+          <div className="w-full lg:w-1/2">
+            <p className="pt-2 text-xs text-left uppercase">
+              css code: <p><span className="font-bold">{cssCode || " -- "}</span></p>
```
```suggestion
              css code: <p><span className="font-bold normal-case">{cssCode || " -- "}</span></p>
```

joekotvas reviewed (approved):
We're getting there!

joekotvas committed:
Apply suggestions from code review

Turns out html character codes and entities are case sensitive. This patch should restore the codes to the proper case and allow long html codes to break properly.
